relays/client-substrate/src/client.rs

```diff
@@ -397,10 +426,8 @@ impl<C: Chain> Client<C> {
 impl JustificationsSubscription {
 	/// Return next justification from the subscription.
 	pub async fn next(&self) -> Result<Option<Bytes>> {
-		let subscription = self.1.clone();
-		self.0
-			.spawn(async move { subscription.lock().await.next().await })
-			.await?
-			.map_err(Error::RpcError)
+		let mut receiver = self.0.lock().await;
+		let justification = receiver.next().await;
```

Comment:
Does it mean we keep the `lock` active when we wait for the next justification? Why do we have multiple tasks for that if only one of them is really making progress at the same time anyway? (i.e. why is it `Arc<Mutex` and we have multiple `JustificationSubscription` instances).

Reply:
We don't have multiple `JustificationSubscription` instances. `JustificationSubscription` needs to be `Send + Sync` => `Arc<Mutex<>>`. It is polled using `now_or_never()`, so if either `lock()` or `next()` are not `Poll::Ready(_)`, it'll immediately return

Reply:
I see. Although `Mutex` seems to be `Send+Sync` already, I guess we can now drop the `Arc` since we never move the mutex to a different future (as in previous code).